### DISCUSSION TITLE
Git storage target hosted on Azure DevOps

### DISCUSSION THREAD

**Discussion by valeriocapponi** [requarks/wiki#5403] [Error / Bug Report]
Hello,

I have a git repo in Azure Devops. 
I have successfully created my key pair and I can use the key to connect to the repo, it is also initialized and has some content in it. However in wiki I have this error:

`
Unable to negotiate with x.x.x.x port 22: no matching host key type found. Their offer: ssh-rsa fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists. error: Could not fetch origin
`

But if I check on the logs I can see:

`
2022-06-14T13:16:29.712653556Z: [INFO]  2022-06-14T13:16:29.712Z [MASTER] warn: fatal: ambiguous argument 'main': unknown revision or path not in the working tree.
2022-06-14T13:16:29.712691456Z: [INFO]  Use '--' to separate paths from revisions, like this:
2022-06-14T13:16:29.712699456Z: [INFO]  'git <command> [<revision>...] -- [<file>...]'
`

main is the only branch I have on the repo. Any ideas?

**Comment by Just-Insane** [comment]
Where you able to find a solution to this? I had it set up and working using Docker. However, it seems to have broken about a month ago. No changes other than some updates were performed on the host.

**Comment by thomas-mc-work** [comment]
I had to initialize my new and empty remote repository by adding a file and thus creating the branch. After that, the synchronization worked fine.
